config/initializers/diaspora_federation.rb

```diff
@@ -61,5 +35,30 @@
 
       person_entity.save!
     end
+
+    on :fetch_private_key_by_id do |diaspora_id|
+      key = Person.where(diaspora_handle: diaspora_id).joins(:owner).pluck(:serialized_private_key).first
+      OpenSSL::PKey::RSA.new key unless key.nil?
+    end
+
+    on :fetch_private_key_by_post_guid do |guid|
+      key = Post.where(guid: guid).joins(author: :owner).pluck(:serialized_private_key).first
+      OpenSSL::PKey::RSA.new key unless key.nil?
+    end
+
+    on :fetch_public_key_by_id do |diaspora_id|
+      key = Person.where(diaspora_handle: diaspora_id).pluck(:serialized_public_key).first
+      OpenSSL::PKey::RSA.new key unless key.nil?
+    end
+
+    on :fetch_public_key_by_post_guid do |guid|
+      key = Post.where(guid: guid).joins(:author).pluck(:serialized_public_key).first
+      OpenSSL::PKey::RSA.new key unless key.nil?
+    end
+
+    on :post_author_is_local? do |guid|
+      id_array = Post.where(guid: guid).joins(:author).pluck(:owner_id)
+      !id_array.first.nil? unless id_array.empty?
```

Comment:
Look at the [spec](https://github.com/cmrd-senya/diaspora/commit/cc821e19744ef57a8155763bb2e5bee06f0b2699#diff-89118032a5e2b114d8286b81e0db3b02R235). There is three different cases, and I made it to return nil if guid is not found. With what you offered it would be false in that case, which isn't true strictly.